packages/zoe/src/contractFacet/seat.js

```diff
@@ -0,0 +1,107 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { assert, details } from '@agoric/assert';
+
+import { isOfferSafe } from './offerSafety';
+
+import '../../exported';
+import '../internal-types';
+
+/** @type {MakeZcfSeatAdminKit} */
+export const makeZcfSeatAdminKit = (
+  allSeatStagings,
+  zoeSeatAdmin,
+  seatData,
+  getAmountMath,
+) => {
+  // The proposal and notifier are not reassigned.
+  const { proposal, notifier } = seatData;
+
+  // The currentAllocation, exited, and stagedAllocation may be reassigned.
+  let currentAllocation = harden(seatData.initialAllocation);
+  let exited = false; // seat is "active"
+  const assertExitedFalse = () => assert(!exited, `seat has been exited`);
+
+  /** @type {ZCFSeatAdmin} */
+  const zcfSeatAdmin = harden({
+    commit: seatStaging => {
+      assertExitedFalse();
+      assert(
+        allSeatStagings.has(seatStaging),
+        details`The seatStaging ${seatStaging} was not recognized`,
+      );
+      currentAllocation = seatStaging.getStagedAllocation();
+      E(zoeSeatAdmin).replaceAllocation(currentAllocation);
+    },
+    updateHasExited: () => {
+      assertExitedFalse();
+      exited = true;
+    },
+  });
+
+  /** @type {ZCFSeat} */
+  const zcfSeat = harden({
+    exit: () => {
+      assertExitedFalse();
+      zcfSeatAdmin.updateHasExited();
+      E(zoeSeatAdmin).exit();
+    },
+    kickOut: (msg = 'Kicked out of seat') => {
+      assertExitedFalse();
+      zcfSeat.exit();
+      assert.fail(msg);
+    },
+    getNotifier: () => {
+      assertExitedFalse();
+      return notifier;
+    },
+    hasExited: () => exited,
```

Comment:
After the zcfSeat has exited, it is not completely inoperative --- its `hasExited` method still responds with an answer appropriate for the concept of an "exited seat". Some of these other methods could also give answers appropriate for an exited seat. `getNotifier` in particular could just return the notifier, which presumably is already set to report some terminated state (finished or failed).

There's no reason for `getProposal` to stop returning the `proposal` which is immutable anyway. It is still the proposal that this seat made. etc

`getAmountAllocated`, `getCurrentAmount`, and `stage` should still refuse. I'm on the fence about `isOfferSafe` but leaning towards having it continue to refuse.